src/module_2/Module_2_EDA_Ex1.md

```diff
@@ -0,0 +1,3152 @@
+plt.xticks(ticks=range(len(monthly_sales)), labels=[f'{year}-{month:02d}' for year, month in monthly_sales.index], rotation=90)
+plt.show()
+```
+
+
+    
+![png](Module_2_EDA_Ex1_files/Module_2_EDA_Ex1_100_0.png)
+    
+
+
+We can observe that the last months of the year are the ones when more products are sold. Furthermore, in January and February of 2022 we can observe that the company arrived to the most successful month, with more than 8000 products sold.
+
+Something that caughts my attention is how the number of products sold increases drastically each year, indicating that, probably, the strategies been performed by the company are flourishing.
+
+**b. How many products are abandoned**
+
+To get more insights, I will plot the number of prodcuts abandoned each month, together with the number of products that are indeed sold.
+
+
+```python
+data['product_abandoned_count'] = data['abandoned_items'].apply(len)
+```
+
+    /tmp/ipykernel_154450/3027958918.py:1: SettingWithCopyWarning: 
+    A value is trying to be set on a copy of a slice from a DataFrame.
+    Try using .loc[row_indexer,col_indexer] = value instead
+    
+    See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
+      data['product_abandoned_count'] = data['abandoned_items'].apply(len)
+
+
+
+```python
+# Aggregate counts by order year and month
+grouped_data = data.groupby([data['order_date'].dt.year, data['order_date'].dt.month])
+monthly_data = grouped_data.agg({
+    'product_order_count' : 'sum',
+    'product_abandoned_count' : sum
+    })
+
+for year in monthly_sales.index.get_level_values(0).unique():
+    monthly_data.loc[year].plot(kind='bar', stacked = True)
+    plt.title(f'Number of Products Sold and Abandoned Each Month in the year {year}')
+    plt.xlabel('Order Date')
+    plt.ylabel('Products Sold')
+    plt.xticks(ticks=range(len(monthly_sales.loc[year])), labels=[f'{year}-{month:02d}' for month in monthly_sales.loc[year].index], rotation=45)
+    plt.legend(['Ordered products', 'Abandoned products'])
+    plt.show()
+```
+
+    /tmp/ipykernel_154450/1745198578.py:3: FutureWarning: The provided callable <built-in function sum> is currently using SeriesGroupBy.sum. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "sum" instead.
+      monthly_data = grouped_data.agg({
+
+
+
+    
+![png](Module_2_EDA_Ex1_files/Module_2_EDA_Ex1_104_1.png)
+    
+
+
+
+    
+![png](Module_2_EDA_Ex1_files/Module_2_EDA_Ex1_104_2.png)
+    
+
+
+
+    
+![png](Module_2_EDA_Ex1_files/Module_2_EDA_Ex1_104_3.png)
+    
+
+
+We can observe that, comparing to the number of products that are sold, not that many products are abandoned. Furthermore, we can see that in 2020, there are not many products registered as abandoned. On the other hand, in the last months of 2021, many products were, specially in December.
```

Comment:
These plots are quite bad for this. Diferent scales making them hard to compare. In addition, the interest is the ratio of sold-abandoned. This conclusion you make is not that straightforward without better plotting.